modules/storages/spec/requests/project_storages_open_spec.rb

```diff
@@ -75,7 +75,7 @@
                 get route, {}, { 'HTTP_ACCEPT' => 'text/vnd.turbo-stream.html' }
 
                 expect(last_response.status).to eq (200)
-                expect(last_response.body).to eq ("<turbo-stream action=\"update\" target=\"open-project-storage-modal-body-component\">\n    <template>\n          <div data-view-component=\"true\" class=\"flex-items-center d-flex flex-column\">\n      <div data-view-component=\"true\">      <svg aria-hidden=\"true\" height=\"24\" viewBox=\"0 0 24 24\" version=\"1.1\" width=\"24\" data-view-component=\"true\" class=\"octicon octicon-check-circle color-fg-success\">\n    <path d=\"M17.28 9.28a.75.75 0 0 0-1.06-1.06l-5.97 5.97-2.47-2.47a.75.75 0 0 0-1.06 1.06l3 3a.75.75 0 0 0 1.06 0l6.5-6.5Z\"></path><path d=\"M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Z\"></path>\n</svg>\n</div>\n      <div data-view-component=\"true\">      <span data-view-component=\"true\">Integration setup completed</span>\n</div>\n      <div data-view-component=\"true\">      <span data-view-component=\"true\">You are being redirected</span>\n</div>\n</div>\n\n\n    </template>\n</turbo-stream>\n\n")
+                expect(last_response.body).to eq ("<turbo-stream action=\"update\" target=\"open-project-storage-modal-body-component\">\n    <template>\n        <div data-view-component=\"true\" class=\"flex-items-center p-4 d-flex flex-column\">\n      <div data-view-component=\"true\">      <svg aria-hidden=\"true\" height=\"24\" viewBox=\"0 0 24 24\" version=\"1.1\" width=\"24\" data-view-component=\"true\" class=\"octicon octicon-check-circle color-fg-success\">\n    <path d=\"M17.28 9.28a.75.75 0 0 0-1.06-1.06l-5.97 5.97-2.47-2.47a.75.75 0 0 0-1.06 1.06l3 3a.75.75 0 0 0 1.06 0l6.5-6.5Z\"></path><path d=\"M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Z\"></path>\n</svg>\n</div>\n      <div data-view-component=\"true\">      <h2 data-view-component=\"true\" class=\"text-center\">Integration setup completed</h2>\n</div>\n      <div data-view-component=\"true\">      <span data-view-component=\"true\" class=\"text-center color-fg-muted\">You are being redirected</span>\n</div>\n</div>\n\n\n    </template>\n</turbo-stream>\n\n")
```

Comment:
I think a request/unit test should not compare generated HTML code 1-to-1. That's what we have feature specs for. It would basically mean, that you can't event change some white space or formatting at all, because you would then just update the expected result for the test, without checking if it's actually correct. Because it's to tedious to compare it manually.

I suggest to check that the returned `Content-Type` is `text/vnd.turbo-stream.html` and that the body and content-length are not `0`. The actual "value" should not be checked if it is HTML code.

You could check that you find certain markers in the response body that are stable throughout all layout changes, but that's basically it.

I'm aiming for maintainability and test stability here.

Reply:
As discussed with @ba1ash we leave this test as it is. But at some point if the Prima library changes, the rendered component could look different and this test and potentially other could just fail because of that.

Reply:
As predicted by @apfohl this has happened: https://github.com/opf/openproject/pull/16618/files/183a9db6de11f4aeb98c7325f8df54a55984d4cc#diff-669df77eeda1ce37db271eaa1add6a585036d345bf8190b28dfc9a9edad5294a .